sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -365,10 +365,17 @@ cdef int free_tree(Tree* tree) nogil:
     for i in range(3):
         cnt[i] = 0
     free_recursive(tree, tree.root_node, cnt)
+    if not tree.root_node.is_leaf:
+        free(tree.root_node.children)
+    free(tree.root_node.width)
+    free(tree.root_node.left_edge)
+    free(tree.root_node.center)
+    free(tree.root_node.barycenter)
+    free(tree.root_node.leaf_point_position)
     free(tree.root_node)
```

Comment:
@ogrisel 
Do you mean duplication on these lines ?
https://github.com/AlexanderFabisch/scikit-learn/blob/fix_bhsne/sklearn/manifold/_barnes_hut_tsne.pyx#L399